lib/libzfs/libzfs_util.c

```diff
@@ -782,8 +782,10 @@ zfs_asprintf(libzfs_handle_t *hdl, const char *fmt, ...)
 
 	va_end(ap);
 
-	if (err < 0)
+	if (err < 0) {
 		(void) no_memory(hdl);
+		ret = NULL;
```

Comment:
> Also, zfs_asprintf() and zfs_alloc() say "which will die if the allocation fails", which is strictly false. no_memory() in the same file says "Display an out of memory error message and abort the current program.", which is also wrong, on both counts.

If you trace the call path you'll see this is in fact true when `hdl->libzfs_printerr` is set.  But clearly we can't count on that so I agree setting this to NULL is the right thing to do here.

```
no_memory()
    zfs_error(, EZFS_NOMEM, )
        zfs_error_fmt()
            zfs_verror()
                exit()
```